include/rcpputils/asserts.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCPPUTILS__ASSERTS_HPP_
+#define RCPPUTILS__ASSERTS_HPP_
+
+#include <stdexcept>
+
+namespace rcpputils
+{
+/**
+ * Checks that a state condition passes.
+ *
+ * \param condition
+ * \throw std::runtime_error if the condition is not met.
+ */
+inline void check(bool condition)
+{
+  if (!condition) {
+    throw std::runtime_error{"Check reported invalid state!"};
+  }
+}
+
+/**
+ * Asserts that a condition passes.
+ *
+ * This function behaves similar to assert, except that it throws instead of invoking abort().
+ * \param condition
+ * \throw std::runtime_error if the macro NDEBUG is not set and the condition is not met.
+ */
+inline void ros_assert(bool condition)
```

Comment:
Do we want to prefix this instead of just having it be `rcpputils::assert`? I don't have a strong opinion but the `ros_` prefix does not seem to be well-established in ROS2 as a pattern

Reply:
Addressed in notes at the top. `rcpputils::assert` conflicts with `assert` since its a macro. It might be doable since `std::min` has the same problem.

Reply:
Ah - my bad for not reading thoroughly. SGTM

Reply:
Other C++ assert libraries do something like `namespace::assert_true(condition)` or `namespace::assert::is_true(condition)`. Maybe `rcpputils::assert_true()` is better?

Reply:
I agree with this. I'll change the function signatures to `assert_true` and `check_true`.